app/modules/pipe.c

```diff
@@ -0,0 +1,309 @@
+/*
+** The pipe algo is somewhat similar to the luaL_Buffer algo, except that it uses
+** table to store the LUAL_BUFFERSIZE byte array chunks instead of the stack.
+** Writing is always to the last UD in the table and overflow pushes a new UD to
+** the end of the table.  Reading is always from the first UD in the table and
+** underrun removes the first UD to shift a new one into slot 1.
+**
+** Reads and writes may span multiple UD buffers and if the read spans multiple UDs
+** then the parts are collected as strings on the Lua stack and then concatenated
+** with a `lua_concat()`.
+**
+** Note that pipes also support the undocumented length and tostring operators
+** for debugging puposes, so if p is a pipe then #p[1] gives the effective
+** length of pipe slot 1 and printing p[1] gives its contents
+**
+** Read the docs/modules/pipe.md documentation for a functional description.
+*/
+
+#include "module.h"
+#include "lauxlib.h"
+#include <string.h>
+
+#define INVALID_LEN ((unsigned)-1)
+
+#define LUA_USE_MODULES_PIPE
```

Comment:
I just checked this patch and it seems that this module is enabled by default by this `define`. If its supposed to be enabled by default I think `#define LUA_USE_MODULES_PIPE` in `user_modules.h` should be uncommented and this one removed but I could be wrong.

Reply:
uuUUaaarggh.  Sorry.  I've just added about 2½-3Kb to `dev` builds.  Other than this, no harm done -- unless of course you start to use it.  Yup it should have been disabled.  Though it will need to be enabled by default on my next PR as it will used by `app/lua/lua.c'